hw/ip/pwrmgr/doc/_index.md

```diff
@@ -0,0 +1,256 @@
+
+Once life cycle initialization is done, the fast FSM enables all second level clock gating (see [clock controller](https://docs.google.com/document/d/1j8H9ikPO2TKTRQvRFItjE5myiba1825PD-Yhy-yaLt0/edit?usp=sharing) for more details) and initiates strap sampling.
+For more details on what exactly the strap samples, please see [here](https://docs.google.com/spreadsheets/d/1pH8T1MhQ7TXtP_bFNT85T9jSVIHlxHAfbMnPbsMdjc0/edit?usp=sharing).
+
+Once strap sampling is complete, the system is ready to begin normal operations (note flash initialization is explicitly not done here, please see [sections below](#Flash-Handling) for more details).
+The fast FSM acknowledges the slow FSM (which made the original power up request) and releases the system reset stage - this enables the processor to begin operation.
+Afterwards, the fast FSM transitions to `Active` state and waits for a software low power entry request.
+
+A low power request is initiated by software through a combination of WFI and software low power hint in {{< regref "CONTROL" >}}.
+Specifically, this means if software issues only WFI, the power manager does not treat it as a power down request.
+The notion of WFI is exported from the processor.
+For Ibex, this is currently in the form of `core_sleeping_o`.
+
+In response to the low power entry request, the fast FSM disables all second level clock gating and asserts appropriate resets if required.  The
+fast FSM then requests the slow FSM to take over.
+Once the request is acknowledged, the fast FSM transitions to low power and waits for the next power up request.
+
+
+## Reset Request Handling
+
+There are 3 reset requests in the system - peripheral requested reset such as watchdog, alert escalation reset and non-debug module reset.
+Flash brownout is handled separately and described in [flash handling section](#Flash-Handling) below.
+
+Watchdog and alert escalation resets are handled directly by the power manager, while the non-debug module reset is handled by the reset controller.
+This separation is because the non-debug reset does not affect the life cycle controller, non-volatile storage controllers and alert states.
+There is thus no need to sequence its operation like the others.
+
+The power controller only observes reset requests in two states - the slow FSM `Low_power` state and the fast FSM `Active` state.
+When a reset request is received during slow FSM `Low_power` state, the system begins its usual power up sequence even if a wakeup has not been received.
+
+When a reset request is received during fast FSM `Active` state, the fast FSM asserts resets and transitions back to its `Low_power` state.
+The normal power-up process described [above](#Fast-Clock-Domain-FSM) is then followed to release the resets.
+Note in this case, the slow FSM is “not activated” and remains in its `Idle` state.
```

Comment:
I think we avoid non-ASCII characters (smart quotes).

Reply:
thanks!